File: Quiz App Project/Frontend/QuizApp/src/app/models/CompletedQuiz.ts
import { Quiz } from "./QuizModel";

export class CompletedQuiz {
  constructor(
    public id: string = "",
    public totalScore: number = 0,
    public studentEmail: string = "",
    public quizId: string = "",
    public student: any = null,
    public startedAt: string = "",
    public endedAt: string = "",
    public createdAt: string = "",
    public quizData: Quiz = new Quiz(),
    public creditPoints: number = 0,
  ) {}
}

File: Quiz App Project/Backend/QuizApp/Models/CompletedQuiz.cs
namespace QuizApp.Models
{
    public class CompletedQuiz
    {
        public string Id { get; set; } = string.Empty;
        public int TotalScore { get; set; }
        public string StudentEmail { get; set; } = string.Empty;
        public string QuizId { get; set; } = string.Empty;
        public Student? Student { get; set; }
        public DateTime StartedAt { get; set; }
        public DateTime EndedAt { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public int CreditPoints { get; set; } 
    }
}